File: Algorithms/Selection Sort/Selection_Sort.py
#swap function
def swap(a,b):
    return b,a

#Selection Sort Function
def selection_sort(array,length):
    #first loop
    for i in range (0,length):
        min_idx = i
        #second loop
        for j in range (i+1,length):
            #comparision
            if(array[j] < array[min_idx]):
                min_idx = j
        #calling swap function
        array[i],array[min_idx] = swap(array[i],array[min_idx])
    return array 

#Main function
#empty array creation
array = []
#getting array size from user
length = int(input("Enter the size of array : "))
#loop to append the array items into array
for i in range (length):
    array_item = int(input(f"Enter the item in array position {i+1} : "))
    array.append(array_item)
#calling the bubble sort function and storing the return value in a array
sorted_array = selection_sort(array, length)
#printing the sorted list
print(f"The Sorted Array : {sorted_array}")

#Time Complexity is: O(N^2)

File: Algorithms/Selection Sort/Selection_Sort.cpp
#include<iostream>
#include<stdlib.h>
using namespace std;

//function to print the values in the array
void display_array(int* array, int length)
{
    int i;
    for(i=0;i<length;i++)
    {
        cout<<array[i]<<" ";
    }
}

//Selection Sort Function
int* selection_sort(int* array,int length)
{
    int i,j,temp,min_idx;
    //first loop
    for(i=0;i<length;i++)
    {
        min_idx = i;
        //second loop
        for(j=i+1;j<length;j++)
        {
            //comparision and swapping
            if(array[j] < array[min_idx])
            {
                    min_idx = j;
            }
        }
        temp = array[i];
        array[i] = array[min_idx];
        array[min_idx] = temp;
    }
    return array;
}

//Main function
int main()
{
    int length,element;
    //getting array size from user
    cout<<"Enter the size of array : ";
    cin>>length;
    //empty array creation
    int* array;
    array = (int*)malloc(length*sizeof(int));
    //loop to insert the array items into array
    for(int i=0;i<length;i++)
    {
        cout<<"Enter item no."<<i+1<<" in the array : ";
        cin>>element;
        array[i] = element;
    }
    //empty array creation
    int* sorted_array;
    sorted_array = (int*)malloc(length*sizeof(int));
    //calling the bubble sort function and storing the return value in a array
    sorted_array = selection_sort(array,length);
    //printing the sorted list
    cout<<"The Sorted Array is : ";
    display_array(sorted_array, length);
    return 0;
}

//Time Complexity is: O(N^2)